tests/unit/Hooks/ImageHooksTest.php

```diff
@@ -99,19 +102,18 @@ public function testPostCreate() {
 		$this->logger->log('debug', 'Test Image Height: '.$imageHeight, array('app' => 'MediaMetadata'));
 		$this->logger->log('debug', 'Test Image Width: '.$imageWidth, array('app' => 'MediaMetadata'));
 
-		$this->imageDimension->expects($this->once())
-			->method('setImageId')
-			->with($jpgFile->getId());
-		$this->imageDimension->expects($this->once())
-			->method('setImageHeight')
-			->with($imageHeight);
-		$this->imageDimension->expects($this->once())
-			->method('setImageWidth')
-			->with($imageWidth);
-
-		$this->mapper->expects($this->once())
-			->method('insert')
-			->with($this->imageDimension);
+		$this->db->expects($this->once())
+			->method('lastInsertId')
+			->with($this->equalTo('*PREFIX*table'));
+
+		$this->imageDimension->setId($jpgFile->getId());
+		$this->assertContains('id', $this->imageDimension->getUpdatedFields());
+		$this->imageDimension->setImageHeight($imageHeight);
+		$this->assertEquals(2, count($this->imageDimension->getUpdatedFields()));
```

Comment:
>Actually, right now I had written this only to check if it was working but I had planned to test the values once things start working. 

OK, all good then.

>Also, regarding the TestEntity class, I thought I can test it with the ImageDimension class as it is also an extension of the entity class.

It is in the real class, but in your test it isn't. Methods like `getUpdatedFields` don't exist in your mock class and you won't get any results back.
If I'm right, you should build your ImageDimension mock like TestEntity in the example.